utils/utils.js

```diff
@@ -104,3 +106,87 @@ export function addRepeater(element, title) {
 
   element.append(tag);
 }
+
+export async function decorateTextfield(row, extraOptions) {
+  const miloLibs = getLibs();
+  await Promise.all([
+    import(`${miloLibs}/deps/lit-all.min.js`),
+    import(`${miloLibs}/features/spectrum-web-components/dist/textfield.js`),
+  ]);
+
+  row.classList.add('text-field-row');
+  const cols = row.querySelectorAll(':scope > div');
+  if (!cols.length) return;
+  let placeholderCol;
+  let maxLengthCol;
+  if (cols.length === 1) {
+    [placeholderCol] = cols;
+  } else if (cols.length === 2) {
+    [placeholderCol, maxLengthCol] = cols;
+  }
+  const text = placeholderCol.textContent.trim();
+  const handle = handlize(text);
+
+  const attrTextEl = createTag('div', { class: 'attr-text' }, maxLengthCol.textContent.trim());
+  const maxCharNum = maxLengthCol?.querySelector('strong')?.textContent.trim();
+  const isRequired = attrTextEl.textContent.trim().endsWith('*');
+
+  const input = createTag('sp-textfield', {
+    ...extraOptions,
+    id: `info-field-${handle}`,
```

Comment:
Not sure if you want review already but just a quick note -- I think we should remove predefined IDs from these extracted functions. Similarly I think required, quiet and size are also something that should be defined in extraOptions from higher level.

Reply:
good point. i have made quiet and size as default, so that it can be overridden with options param values.